frontend/app/controllers/spree/orders_controller.rb

```diff
@@ -10,7 +10,8 @@ class OrdersController < Spree::StoreController
     before_action :assign_order_with_lock, only: :update
 
     def show
-      @order = Order.includes(line_items: [variant: [:option_values, :images, :product]], bill_address: :state, ship_address: :state).find_by!(number: params[:id])
+      order_id = params[:id].gsub('-order-success','')
```

Comment:
I wasn't sure if this needs to be moved to separate service to handle adding/removing of '-order-success' to/from id.

Reply:
@damianlegawiec Let me know if this change needs improvement. I will work on fixing the failed tests.